hooks/checkout

```diff
@@ -42,7 +42,10 @@ checkout() {
   if [[ -z "${BUILDKITE_COMMIT:-}" || "${BUILDKITE_COMMIT}" == "HEAD" ]]; then
     git checkout -f FETCH_HEAD
   else
-    git checkout -f "${BUILDKITE_COMMIT}"
+    # If the branch doesn't have the commit we want it might have been force
+    # pushed in the meantime. Exit with ESTALE to signify the stale branch
+    # reference in that case.
+    git checkout -f "${BUILDKITE_COMMIT}" || exit 116
```

Comment:
I think we have to be more precise, e.g. only exit with `116` if we get a `128` in the first place:

```
❯ git checkout -f c0ffee
fatal: reference is not a tree: c0ffee
❯ echo $?
128
```

otherwise it might mask other types of errors that are not due to a stale commit:

```
❯ git checkoutyout -f master
git: 'checkoutyout' is not a git command. See 'git --help'.
❯ echo $?
1
```